hbase_storage/src/main/java/com/splicemachine/si/data/hbase/coprocessor/SIObserver.java

```diff
@@ -193,6 +172,29 @@ public void postCompact(ObserverContext<RegionCoprocessorEnvironment> c, Store s
         }
     }
 
+    private PurgeConfig getPurgeConfig(final FlushLifeCycleTracker tracker, final SConfiguration conf) throws IOException {
+        PurgeConfig purgeConfig;
+        if(tracker instanceof FlushLifeCycleTrackerWithConfig) {
+            purgeConfig = ((FlushLifeCycleTrackerWithConfig) tracker).getConfig();
+        } else {
+            SpliceLogUtils.warn(LOG, "Splice store flusher is not set, as a result flush will ignore minimum retention period " +
+                    "and time travel queries may return inconsistent results. To set Splice flusher review documentation and revise HBase configuration: " +
+                    DefaultStoreEngine.DEFAULT_STORE_FLUSHER_CLASS_KEY);
+
+            PurgeConfigBuilder purgeConfigBuilder = new PurgeConfigBuilder();
+            if (conf.getOlapCompactionAutomaticallyPurgeDeletedRows()) {
+                purgeConfigBuilder.purgeDeletesDuringFlush();
+            } else {
+                purgeConfigBuilder.noPurgeDeletes();
+            }
+            purgeConfigBuilder.transactionLowWatermark(TransactionsWatcher.getLowWatermarkTransaction());
+            purgeConfigBuilder.purgeUpdates(conf.getOlapCompactionAutomaticallyPurgeOldUpdates());
+            purgeConfig = purgeConfigBuilder.build();
+        }
+        assert purgeConfig != null;
+        return purgeConfig;
```

Comment:
I like the fact that we don't break if the SpliceStoreFlusher isn't set, but we do risk having configurations out there that forget to set it for a long period of time. Is it possible to issue a warning as well when calling the SET_MIN_RETENTION_PERIOD system call?

Reply:
done.